### PR TITLE
h265e- use C++ primitives (memcpy => std::copy)

### PR DESCRIPTION
**Pull request by ilapichev**
Issue: Issue: MDP-50973

### CODE REVIEW

**Comment by ilapichev** [comment]
I guess api/include/mfxstructures.h was changed incorrectly.

File: _studio/mfx_lib/encode_hw/h265/include/mfx_h265_encode_hw_utils.h [hunk 1]
```diff
@@ -325,6 +326,12 @@ struct RoiData{
     mfxU32  Bottom;
 
     mfxI16  Priority;
+
+    RoiData& operator = (mfxExtEncoderROI::roi roi)
+    {
+    	*this = roi;
```
Comment (mgonchar) [inline _studio/mfx_lib/encode_hw/h265/include/mfx_h265_encode_hw_utils.h]: what is the point of operator = where you call operator = ?

Remove changes in this file

File: _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw.cpp [hunk 2]
```diff
@@ -1196,7 +1195,7 @@ mfxStatus  MFXVideoENCODEH265_HW::Execute(mfxThreadTask thread_task, mfxU32 /*ui
             {
                 MFX_CHECK(bytesAvailable >= pSEI->DataLength, MFX_ERR_NOT_ENOUGH_BUFFER);
 
-                memcpy_s(bs->Data + bs->DataOffset + bs->DataLength, pSEI->DataLength, pSEI->pData + pSEI->DataOffset, pSEI->DataLength);
+                std::copy(pSEI->pData + pSEI->DataOffset, pSEI->pData + pSEI->DataOffset + pSEI->DataLength / sizeof(UCHAR), bs->Data + bs->DataOffset + bs->DataLength);
```
Comment (mgonchar) [inline _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw.cpp]: you know that sizeof(char) == 1 and everybody else knows, you can omit this

File: _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_bs.cpp [hunk 3]
```diff
@@ -315,7 +315,7 @@ mfxU32 AddEmulationPreventionAndCopy(
 
     if (!bEmulationByteInsertion)
     {
-        memcpy_s(bsDataStart, len, sbegin, len);
+        std::copy(sbegin, sbegin + len / sizeof(mfxU8), bsDataStart);
```
Comment (mgonchar) [inline _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_bs.cpp]: same

File: _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_bs.cpp [hunk 4]
```diff
@@ -396,7 +396,7 @@ void BitstreamWriter::PutBitsBuffer(mfxU32 n, void* bb, mfxU32 o)
         n &= 7;
 
         assert(N + !!n < (m_bsEnd - m_bs));
-        memcpy_s(m_bs, N, b, N);
+        std::copy(b, b + N / sizeof(mfxU8), m_bs);
```
Comment (mgonchar) [inline _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_bs.cpp]: same

File: _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_utils.cpp [hunk 5]
```diff
@@ -3461,7 +3462,7 @@ void ConfigureTask(
     {
         for (mfxU16 i = 0; i < parRoi->NumROI; i ++)
         {
-            memcpy_s(&task.m_roi[i], sizeof(RoiData), &parRoi->ROI[i], sizeof(RoiData));
+        	task.m_roi[i] = parRoi->ROI[i];
```
Comment (mgonchar) [inline _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_utils.cpp]: see https://github.com/Intel-Media-SDK/MediaSDK/pull/724 to get the idea of how to do this without custom assignment operator

File: api/include/mfxstructures.h [hunk 6]
```diff
@@ -1426,7 +1426,7 @@ typedef struct {
     mfxU16  ROIMode;
     mfxU16  reserved1[10];
 
-    struct  {
+    struct roi {
```
Comment (mgonchar) [inline api/include/mfxstructures.h]: first:
you don't need several lines:
```
 struct roi {
...
} ROI[256];
```

second:
don't change API please, it is not allowed without changing API version

Comment (ilapichev) [reply api/include/mfxstructures.h]: ok

**Comment by dvrogozh** [comment]
@ilapichev: you are based on outdated forked copy, that's why tests are failing. Please, rebase your patch!

**Pull request closed without merging**